solver/src/interactions/uniswap.rs

```diff
@@ -27,32 +27,34 @@ impl Interaction for UniswapInteraction {
 impl UniswapInteraction {
     fn encode_approve(&self) -> EncodedInteraction {
         let token = ERC20::at(&self.web3(), self.token_in);
-        let method = token.approve(self.contract.address(), U256::MAX);
+        let method = token.approve(self.router.address(), U256::MAX);
         let calldata = method.tx.data.expect("no calldata").0;
         (self.token_in, 0.into(), calldata)
     }
 
     fn encode_swap(&self) -> EncodedInteraction {
-        let method = self.contract.swap_exact_tokens_for_tokens(
+        let method = self.router.swap_exact_tokens_for_tokens(
             self.amount_in,
             self.amount_out_min,
             vec![self.token_in, self.token_out],
             self.settlement.address(),
             U256::MAX,
         );
         let calldata = method.tx.data.expect("no calldata").0;
-        (self.contract.address(), 0.into(), calldata)
+        (self.router.address(), 0.into(), calldata)
     }
 
     fn web3(&self) -> web3::Web3<ethcontract::transport::DynTransport> {
-        self.contract.raw_instance().web3()
+        // TODO - This is a cheeky way of fetching web3
+        self.settlement.raw_instance().web3()
```

Comment:
It's fine imo. Can keep it this way.

Reply:
Should I remove the TODO?

Reply:
```suggestion
        self.settlement.raw_instance().web3()
```

Reply:
If you agree that it's fine I'd say remove it.